package/MDAnalysis/analysis/results.py

```diff
@@ -0,0 +1,310 @@
+
+
+class ResultsGroup:
+    """
+    Management and aggregation of results stored in :class:`Results` instances.
+
+    A :class:`ResultsGroup` is an optional description for :class:`Result` "dictionaries"
+    that are used in analysis classes based on :class:`AnalysisBase`. For each *key* in a
+    :class:`Result` it describes how multiple pieces of the data held under the key are
+    to be aggregated. This approach is necessary when parts of a trajectory are analyzed
+    independently (e.g., in parallel) and then need to me merged (with :meth:`merge`) to
+    obtain a complete data set.
+
+    Parameters
+    ----------
+    lookup : dict[str, Callable], optional
+        aggregation functions lookup dict, by default None
+
+    Examples
+    --------
+
+    .. code-block:: python
+
+        from MDAnalysis.analysis.results import ResultsGroup, Results
+        group = ResultsGroup(lookup={'mass': ResultsGroup.float_mean})
+        obj1 = Results(mass=1)
+        obj2 = Results(mass=3)
+        assert {'mass': 2.0} == group.merge([obj1, obj2])
+    
+
+    .. code-block:: python
+
+        # you can also set `None` for those attributes that you want to skip
+        lookup = {'mass': ResultsGroup.float_mean, 'trajectory': None}
+        group = ResultsGroup(lookup)
+        objects = [Results(mass=1, skip=None), Results(mass=3, skip=object)]
+        assert group.merge(objects, require_all_aggregators=False) == {'mass': 2.0}
+
+    .. versionadded:: 2.8.0
+    """
+
+    def __init__(self, lookup: dict[str, Callable] = None):
+        self._lookup = lookup
+
+    def merge(self, objects: Sequence[Results], require_all_aggregators: bool = True) -> Results:
+        """Merge results into a single object. If objects contain single element, returns it while ignoring _lookup attribute.
+
+        Parameters
+        ----------
+        require_all_aggregators : bool, optional
+            if True, raise an exception when no aggregation function for a particular argument is found.
+            Allows to skip aggregation for the parameters that aren't needed in the final object -- see :class:`ResultsGroup`.
+
+        Returns
+        -------
+        Results
+            merged :class:`Results`
+
+        Raises
+        ------
+        ValueError
+            if no aggregation function for a key is found and ``require_all_aggregators=True``
+        """
+        if len(objects) == 1:
+            merged_results = objects[0]
+        else:
+            merged_results = Results()
+            for key in objects[0].keys():
+                agg_function = self._lookup.get(key, None)
+                if agg_function is not None:
+                    results_of_t = [obj[key] for obj in objects]
+                    merged_results[key] = agg_function(results_of_t)
+                else:
+                    if require_all_aggregators:
+                        raise ValueError(f"No aggregation function for {key=}")
+        return merged_results
```

Comment:
```suggestion
        if len(objects) == 1:
            merged_results = objects[0]
            return merged_results
        
        merged_results = Results()
        for key in objects[0].keys():
            agg_function = self._lookup.get(key, None)
            if agg_function is not None:
                results_of_t = [obj[key] for obj in objects]
                merged_results[key] = agg_function(results_of_t)
            elif require_all_aggregators:
                raise ValueError(f"No aggregation function for {key=}")
        return merged_results
```
Just to reduce some of the indentation

Reply:
Indeed, thanks.